onnxscript/ir/_protocols.py

```diff
@@ -14,8 +14,8 @@
 
 import typing
 from typing import (
-    AbstractSet,
```

Comment:
I thought that you were planning to eliminate the protocols?

Reply:
No I wasn't planning to eliminate the protocols. In the last comment I made I decided to not separate the read-only protocols from the mutable protocols because if we do so they will have to be used together and conflict.

Here all protocols define mutation methods so we are good.